starter-code/app.js

```diff
@@ -14,24 +15,82 @@ const siteController = require("./routes/siteController");
 // Mongoose configuration
 mongoose.connect("mongodb://localhost/ibi-ironhack");
 
+const User = require("./models/user");
+const session = require("express-session");
+const bcrypt = require("bcrypt");
+const passport = require("passport");
+const LocalStrategy = require("passport-local").Strategy;
+const flash = require("connect-flash");
+
+//enable sessions here
+app.use(
+  session({
+    secret: "our-passport-local-strategy-app",
+    resave: true,
+    saveUninitialized: true
+  })
+);
+//initialize passport and session here
+app.use(passport.initialize());
+app.use(passport.session());
+app.use(flash());
 // view engine setup
-app.set('views', path.join(__dirname, 'views'));
-app.set('view engine', 'ejs');
+app.use(expressLayouts);
+app.set("layout", "layouts/main-layout");
+app.set("views", path.join(__dirname, "views"));
+app.set("view engine", "ejs");
 
 // uncomment after placing your favicon in /public
 //app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
-app.use(logger('combined'));
+app.use(logger("combined"));
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(cookieParser());
-app.use(express.static(path.join(__dirname, 'public')));
+app.use(express.static(path.join(__dirname, "public")));
 
 // Routes
 app.use("/", siteController);
```

Comment:
Your controller/router should always come *after* all the configuration for middlewares, otherwise you are navigating before all that code is executed. Make sure you move this `use` line to after the passport config, right before the error handlers.